CPUMeter.c

```diff
@@ -38,25 +38,41 @@ static void CPUMeter_updateValues(Meter* this, char* buffer, int size) {
       return;
    }
    memset(this->values, 0, sizeof(double) * CPU_METER_ITEMCOUNT);
-   double percent = Platform_setCPUValues(this, cpu);
+
+   char cpuUsageBuffer[8] = { 0 };
+   char cpuFrequencyBuffer[16] = { 0 };
+   char cpuTemperatureBuffer[10] = { 0 };
+
+   if (this->pl->settings->showCPUUsage) {
+      double percent = Platform_setCPUValues(this, cpu);
+      xSnprintf(cpuUsageBuffer, sizeof(cpuUsageBuffer), "%5.1f%%", percent);
+   }
+
    if (this->pl->settings->showCPUFrequency) {
       double cpuFrequency = this->values[CPU_METER_FREQUENCY];
-      char cpuFrequencyBuffer[16];
       if (cpuFrequency < 0) {
          xSnprintf(cpuFrequencyBuffer, sizeof(cpuFrequencyBuffer), "N/A");
       } else {
          xSnprintf(cpuFrequencyBuffer, sizeof(cpuFrequencyBuffer), "%.0fMHz", cpuFrequency);
       }
-      if (this->pl->settings->showCPUUsage) {
-         xSnprintf(buffer, size, "%5.1f%% %s", percent, cpuFrequencyBuffer);
-      } else {
-         xSnprintf(buffer, size, "%s", cpuFrequencyBuffer);
-      }
-   } else if (this->pl->settings->showCPUUsage) {
-      xSnprintf(buffer, size, "%5.1f%%", percent);
-   } else if (size > 0) {
-      buffer[0] = '\0';
    }
+
+   if (this->pl->settings->showCPUTemperature) {
+      double cpuTemperature = this->values[CPU_METER_TEMPERATURE];
+      if (cpuTemperature < 0)
```

Comment:
What about very cold rooms? Prefer to use NAN for unavailable values. Cf. #101 for a similar change for the frequency stuff.

Reply:
converted